File: src/TinyHttpRequest.scala

class TinyHttpRequest(inputArray: Array[String]) {
  var method: String = null
  var path: String = null
  var headers: Map[String, String] = Map()

  createFromRequestMessage(inputArray)

  def createFromRequestMessage(inputArray: Array[String]): Unit = {
    if (inputArray.size == 0) {
      return;
    }

    val initialRequestLine = inputArray(0)
    val initialLineSegments: Array[String] = initialRequestLine.split(" ")

    this.method = initialLineSegments(0)
    this.path = Constants.ROOT_PATH + "/" + initialLineSegments(1)


    inputArray.slice(1, inputArray.length).foreach(line => {
      val segments = line.split(": ")
      headers += segments(0) -> segments(1)
    })
  }

  override def toString = s"TinyHttpRequest: $method"
}

File: src/tiny-http-request.js
const ROOT_PATH = './static';

export class TinyHttpRequest {
    constructor (requestData) {
        const requestBody = requestData.toString().split('\r\n');
        const initialRequestLine = requestBody[0];
        const initialLineSegs = initialRequestLine.split(' ');

        this.method = initialLineSegs[0];
        this.path = ROOT_PATH + initialLineSegs[1];
        this.headers = {};

        const headerLines = requestBody.slice(1);
        for (let line of headerLines) {
            if (line === '') {
                continue;
            }

            const segs = line.split(': ');

            this.headers[segs[0]] = segs[1];
        }
    }
}
